client-runtime/src/main/java/com/microsoft/rest/RestClient.java

```diff
@@ -194,7 +196,19 @@ private Builder(final RestClient restClient) {
         /**
          * Creates an instance of the builder.
          */
-        public Builder() { }
+        public Builder() {
+            this.httpClientFactory = new RxNettyAdapter.Factory();
```

Comment:
This line concerns me. I want to have a reasonable way of adding the required parameters to a builder, and a reasonable way of referencing the concrete dependencies.

It might wind up being best to just create this default factory when building if the given factory is null.